File: lib/debate/models/topic_model.dart
class TopicModel {
  final String id;
  final String title;
  final String description;

  TopicModel({
    required this.id,
    required this.title,
    required this.description,
  });

  factory TopicModel.fromJson(Map<String, dynamic> json) {
    return TopicModel(
      id: json['_id'] ?? '', // Mongoose uses _id
      title: json['title'] ?? '',
      description: json['description'] ?? '',
    );
  }
}

File: server/src/modules/topic/topic.model.js
const mongoose = require('mongoose');
// Plugins import removed as the file does not exist.
// const { toJSON, paginate } = require('../../models/plugins');

const topicSchema = mongoose.Schema(
    {
        title: {
            type: String,
            required: true,
            trim: true,
        },
        description: {
            type: String,
            required: true,
            trim: true,
        },
        createdBy: {
            type: String,
            required: true,
        },
        isActive: {
            type: Boolean,
            default: true,
        },
    },
    {
        timestamps: true,
    }
);

// topicSchema.plugin(toJSON);
// topicSchema.plugin(paginate);

const Topic = mongoose.model('Topic', topicSchema);

module.exports = Topic;
